### DISCUSSION TITLE
Maintain spacing between header and tables/code blocks

### DISCUSSION THREAD

**Discussion by cmason3** [jgm/pandoc#9326] [Q&A]
I am using Pandoc to convert from HTML to DOCX via a reference template. Everything is working fine, but I am unable to get it to maintain space between a heading and a table or code block element. If I look at the style of the heading element then the spacing is added before and not after.

Do any Pandoc/Word experts know if it is possible to add spacing to the start of a table or code block as I am unable to find an option to do this in the Word Style?

![image](https://github.com/jgm/pandoc/assets/57402469/4b647e7b-f671-4719-9290-0ccd87fe6dc3)

**Comment by jgm** [comment]
If you figure it out, let me know -- I think it could be a useful addition to the default style we use.

**Comment by cmason3** [comment]
I think I have managed to fix it by playing around with different style settings:

![image](https://github.com/jgm/pandoc/assets/57402469/444abf0c-e6bd-4b40-b598-0fad0dfdc688)

In my reference template, I updated the style for all the Heading styles to include the same Spacing Before and After and then ticked the "Don't add space between paragraphs of the same style" and it appears to have worked:

![image](https://github.com/jgm/pandoc/assets/57402469/f36525c7-c981-45a7-ab59-d6cde8c1cbad)

**Reply by cmason3** [reply]
Actually you don't want to tick that box else you will lose the spacing if you have a Heading after a Heading. I incorrectly assumed that if we applied Spacing After then we would have double spacing between a Heading and a Paragraph as it would add the Heading Spacing After and the Paragraph Spacing Before together, but it seems to be cleverer than that.
